tests/test_teacher.py

```diff
@@ -54,33 +55,40 @@ def test_GroundAtomsHold():
 def test_DemonstrationQuery():
     """Tests for answering queries of type DemonstrationQuery."""
     utils.reset_config({"env": "cover", "approach": "unittest"})
-    teacher = Teacher()
     env = create_env("cover")
-    task = env.get_train_tasks()[0]
+    train_tasks = env.get_train_tasks()
+    teacher = Teacher(train_tasks)
+    train_task_idx = 0
+    task = train_tasks[train_task_idx]
     state = task.init
     goal = task.goal
     # Test normal usage
-    query = DemonstrationQuery(goal)
+    query = DemonstrationQuery(train_task_idx)
     response = teacher.answer_query(state, query)
     assert isinstance(response, DemonstrationResponse)
     assert response.query is query
     assert isinstance(response.teacher_traj, LowLevelTrajectory)
     assert len(response.teacher_traj.actions) == 2
     assert all(atom.holds(response.teacher_traj.states[-1]) for atom in goal)
+    assert response.teacher_traj.is_demo
+    assert response.teacher_traj.train_task_idx == train_task_idx
     # Test usage when goal is already achieved
     response = teacher.answer_query(response.teacher_traj.states[-1], query)
     assert isinstance(response, DemonstrationResponse)
     assert response.query is query
     assert isinstance(response.teacher_traj, LowLevelTrajectory)
     assert len(response.teacher_traj.actions) == 0
     # Test usage when achieving goal is impossible
-    block_type = [t for t in env.types if t.name == "block"][0]
-    block = block_type("block0")
+    state = train_tasks[0].init
+    block = [b for b in state if b.name == "block1"][0]
```

Comment:
does it matter that this was block0 before but block1 now?

Reply:
no, but i changed it back to reduce entropy in the cosmos